### PR TITLE
feat(sql): Provided module

### PR DESCRIPTION
**Pull request by ekkinox**
Yokai SQL module:
- based on `database/sql`
- support of `mysql`, `postgres` and `sqlite`
- with a hooking mechanism decorating `Conn`, `Tx`, `Stmt`, etc
- with 2 ready to use hooks: for `logging` (yokai logs), `tracing` (yokai OTEL traces)

### CODE REVIEW

File: sql/hook/util_test.go [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+func TestContains(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		name string
+		list []string
+		item string
+		want bool
+	}{
+		{
```
Comment (ndk) [inline sql/hook/util_test.go]: My general suggestion is always to cover corner cases such as a nil slice, an empty slice, or a slice with one item. :)

Comment (ekkinox) [reply sql/hook/util_test.go]: Will push this in next commit 👍

File: sql/registry.go [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+func NewDefaultDriverRegistry() *DefaultDriverRegistry {
+	return &DefaultDriverRegistry{
+		drivers: make(map[string]*Driver),
+	}
+}
+
+// Add adds and register a given Driver for a name.
+func (r *DefaultDriverRegistry) Add(name string, driver *Driver) (err error) {
+	if r.Has(name) {
+		return err
```
Comment (ndk) [inline sql/registry.go]: If it returns `nil` for already registered drivers on purpose, it would be less confusing if it returns `nil` explicitly.
```suggestion
		return nil
```

File: sql/system_test.go [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+		},
+		{
+			"mysql",
+			sql.MysqlSystem,
+		},
+		{
+			"postgres",
+			sql.PostgresSystem,
+		},
+		{
```
Comment (ndk) [inline sql/system_test.go:67]:
```suggestion
		{
			"MySqL",
			sql.MysqlSystem,
		},
		{
```

File: sql/convert.go [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+	"database/sql/driver"
+)
+
+// ConvertNamedValuesToValues converts a list of driver.NamedValue into a list of driver.Value.
+func ConvertNamedValuesToValues(namedValues []driver.NamedValue) []driver.Value {
+	values := make([]driver.Value, len(namedValues))
+
+	for name, value := range namedValues {
+		values[name] = value.Value
+	}
```
Comment (ndk) [inline sql/convert.go]: A bit confusing :)
```suggestion
	for i, value := range namedValues {
		values[i] = value.Value
	}
```

**Review by ndk — APPROVED** [review]
I've left some minor suggestions, but overall, LGTM.

File: sql/registry.go (hunk 2 above)
Comment (ekkinox) [reply sql/registry.go]: agreed :+1:  done

File: sql/system_test.go (hunk 3 above)
Comment (ekkinox) [reply sql/system_test.go:67]: done, and added more similar cases

File: sql/convert.go (hunk 4 above)
Comment (ekkinox) [reply sql/convert.go]: done :)

**Review by ndk — APPROVED** [review]
